.github/workflows/deploy-stg-enhanced-worker.yml

```diff
@@ -0,0 +1,32 @@
+on:
+  push:
+    paths: ["src/**"]
+    branches:
+      - "master"
+      - "integration"
+      - 'DOP-3833'
+
+concurrency:
+  group: environment-stg-${{ github.ref }}
```

Comment:
Would having the same group name here conflict with the workflow for the webhook deployment (basing this question off of this [docs example](https://docs.github.com/en/actions/using-jobs/using-concurrency#example-only-cancel-in-progress-jobs-or-runs-for-the-current-workflow))? For example, if a new PR changes files in paths for both the worker and the API, I'm wondering if this might unintentionally cause one of the workflows to get cancelled